simulator/option_manager.go

```diff
@@ -70,6 +78,7 @@ func (m *OptionManager) QueryOptions(req *types.QueryOptions) soap.HasFault {
 	return body
 }
 
+// find requires holding the lock.
```

Comment:
Either:

* Update the GoDoc to be more clear:
    ```go
    // find is not thread-safe and requires callers to already hold either
    // a shared or exclusive lock from the OptionsManager
    ```

**or**

* Move the function into `UpdateOptions` so only it can invoke `find`, ex:
    ```go
    func (m *OptionManager) UpdateOptions(req *types.UpdateOptions) soap.HasFault {
    	m.Lock()
    	defer m.Unlock()
    
    	find := func(key string) *types.OptionValue {
    		for _, opt := range m.Setting {
    			setting := opt.GetOptionValue()
    			if setting.Key == key {
    				return setting
    			}
    		}
    		return nil
    	}
    
    	// ...
    ```